pyxem/signals/beam_shift.py

```diff
@@ -62,42 +75,507 @@ def make_linear_plane(self, mask=None):
         mask : HyperSpy signal, optional
             Must be the same shape as the navigation dimensions of the beam
             shift signal. The True values will be masked.
+        fit_corners : float, optional
+            Make a mask so that the planes are fitted to the corners of the
+            signal. This mush be set with a number, like 0.05 (5%) or 0.10 (10%).
 
         Examples
         --------
         >>> s = pxm.signals.BeamShift(np.random.randint(0, 99, (100, 120, 2)))
         >>> s_mask = hs.signals.Signal2D(np.zeros((100, 120), dtype=bool))
         >>> s_mask.data[20:-20, 20:-20] = True
-        >>> s.make_linear_plane(mask=s_mask)
+        >>> s.get_linear_plane(mask=s_mask)
 
         """
         if self._lazy:
             raise ValueError(
                 "make_linear_plane is not implemented for lazy signals, "
                 "run compute() first"
             )
+        if (mask is not None) and (fit_corners is not None):
+            raise ValueError("Only mask or fit_to_corners can be set.")
+        if fit_corners is not None:
+            mask = bst._get_corner_mask(self.isig[0], corner_size=fit_corners)
+
```

Comment:
This could be more efficient. Right now this requires that you calculate the entire shifts when really we only need to calculate the corners.